### PR TITLE
Migrate RandomCrop to Keras CV

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
https://github.com/keras-team/keras-cv/pull/1134

### CODE REVIEW

File: keras_cv/layers/preprocessing/random_crop.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+
+# In order to support both unbatched and batched inputs, the horizontal
+# and verticle axis is reverse indexed
+H_AXIS = -3
+W_AXIS = -2
+IMAGES = "images"
+BOUNDING_BOXES = "bounding_boxes"
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop.py:27]: Looks like these string constants are unused

File: keras_cv/layers/preprocessing/random_crop_test.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+import numpy as np
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers.preprocessing.random_crop import RandomCrop
+
+
+class RandomCropTest(tf.test.TestCase, parameterized.TestCase):
+    def _run_test(self, expected_height, expected_width):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop_test.py:24]: Same comment as before re: naming this test_output_shape and parameterizing directly

**Review by ianstenbit — CHANGES_REQUESTED** [review]

File: keras_cv/layers/preprocessing/random_crop.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop.py]: removed

File: keras_cv/layers/preprocessing/random_crop_test.py (hunk 2 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop_test.py]: updated

**Review by ianstenbit — APPROVED** [review]

**Pull request merged**